languages/multi-language/src/main/java/de/jplag/multilang/MultiLanguageParser.java

```diff
@@ -29,8 +36,42 @@ public List<Token> parseFiles(Set<File> files, boolean normalize) throws Parsing
         return results;
     }
 
+    private void registerLanguageMaps(List<Language> languages) {
+        for (Language language : languages.stream().sorted(Comparator.comparing(it -> it.hasPriority() ? 1 : 0)).toList()) {
+            for (String suffix : language.suffixes()) {
+                registerLanguageSuffix(language, suffix);
+            }
+
+        }
+    }
+
+    private void registerLanguageSuffix(Language language, String suffix) {
+        if (language.hasPriority()) {
+            registerLanguageSuffixTo(this.languageMapPriority, language, suffix, true);
+        } else {
+            registerLanguageSuffixTo(this.languageMap, language, suffix, !this.languageMapPriority.containsKey(suffix));
+        }
+    }
+
+    private void registerLanguageSuffixTo(Map<String, Language> map, Language language, String suffix, boolean additionalConditionForException) {
+        if (map.containsKey(suffix) && additionalConditionForException) {
+            throw new IllegalStateException(String.format(ERROR_MULTIPLE_LANGUAGES, suffix, language.getName(), map.get(suffix).getName()));
```

Comment:
The condition "more than one non-prioritized language module but no language module with priority" is not necessarily a problem, as a language module with priority might still be found afterwards. Put like this, depending on the order in which the language modules are loaded, an exception might be thrown or not.

Reply:
Concept of alternative implementation:
 - only one `Map<String, Language> languageMap` + one `Set<String> needsPriority`
 - for each `Language language` for each `String fileExtension`:
 - if `!languageMap.containsKey(fileExtension))`: set language for that fileExtension
 - else: check whether `language` and the existing language for `fileExtension` have priority (`newPriority`, `oldPriority`)
 - `newPriority && !oldPriority`: overwrite
 - `newPriority && oldPriority`: throw exception
 - `!newPriority && oldPriority`: do nothing
 - `!newPriority && !oldPriority`: `needsPriority.add(fileExtension)`

In the end, assert that for all file extensions in `needPriority`, a language module with priority has been found, or else throw an exception.